pkg/containers/datasource.go

```diff
@@ -0,0 +1,72 @@
+package containers
+
+import (
+	"encoding/json"
+
+	"github.com/aquasecurity/tracee/types/detect"
+)
+
+type SignaturesDataSource struct {
+	containers *Containers
+}
+
+func NewDataSource(c *Containers) *SignaturesDataSource {
+	return &SignaturesDataSource{
+		containers: c,
+	}
+}
+
+func (ctx SignaturesDataSource) Get(key interface{}) (map[string]interface{}, error) {
+	containerId, ok := key.(string)
+	if !ok {
+		return nil, detect.ErrKeyNotSupported
+	}
+	ctx.containers.mtx.RLock()
+	defer ctx.containers.mtx.RUnlock()
+	for _, cgroup := range ctx.containers.cgroupsMap {
+		if cgroup.Container.ContainerId == containerId {
+			containerData := cgroup.Container
+			podData := containerData.Pod
+			result := make(map[string]interface{}, 7)
+			result["container_id"] = containerData.ContainerId
+			result["container_name"] = containerData.Name
+			result["container_image"] = containerData.Image
+			result["k8s_pod_id"] = podData.UID
+			result["k8s_pod_name"] = podData.Name
+			result["k8s_pod_namespace"] = podData.Namespace
+			result["k8s_pod_sandbox"] = podData.Sandbox
+			return result, nil
+		}
+	}
+	return nil, detect.ErrDataNotFound
+}
+
+func (ctx SignaturesDataSource) Keys() []string {
+	return []string{"string"}
+}
+
+func (ctx SignaturesDataSource) Schema() string {
+	schemaMap := map[string]string{
+		"container_id":      "string",
+		"container_name":    "string",
+		"container_image":   "string",
+		"k8s_pod_id":        "string",
+		"k8s_pod_name":      "string",
+		"k8s_pod_namespace": "string",
+		"k8s_pod_sandbox":   "bool",
+	}
+	schema, _ := json.Marshal(schemaMap)
+	return string(schema)
+}
+
+func (ctx SignaturesDataSource) Version() uint {
+	return 1
```

Comment:
WDYT about standartazing YYYYMMDDHHMM (or anything that can be monotonically increased whenever changed) ? This idea comes from the DNS SOA serial number (there is a RFC describing that each time the zone config changes the serial number should be increased to that format, or similar, so you are always increasing numbers AND, at the same time, you know the last time of change (in our case of the "format").

Reply:
So instead of the `int` we would return a `time.Time` and then signature consumers would compare to that date? The versioning is a bit more descriptive but not sure if it would be as user friendly. On the fence for this.

Reply:
So instead we would check `Version > 1684936308 => fail` instead of `Version > 1 => fail`? I guess we could except we would have to document it a bit more specifically. 
Still on the fence as this requires _slightly_ more effort to parse and when updating sources, instead of just incrementing the counter. So what you suggest has more info encoded but we could encode that info somewhere else in the data source as well.

Reply:
TODO: check if you want to address this after freeze (freezing tomorrow). This will be merged as is by https://github.com/aquasecurity/tracee/pull/3142

Reply:
I thought about semantic versining (major-minor) where major is increased in breaking changes (schema changed) and minor is increased in addition of new fields. Then, a signature keeps working as long as the data source has the same major, and when it adds new fields, only signatures that look for a new field will also check if the minor is big enough.